cw/capture.py

```diff
@@ -1039,80 +1102,85 @@ def capture_ecdsa_simple(ot, fw_bin, pll_frequency, capture_cfg):
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            ])
+            priv_key_d = priv_key_d0 + priv_key_d1
+            # Set two shares of the scalar secret_k
+            secret_k0 = bytearray([
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            ])
+            secret_k1 = bytearray([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            ])
+            secret_k = secret_k0 + secret_k1
+        # ECDSA-256
+        elif capture_cfg["key_len_bytes"] == 32:
+            # Set two shares of the private key d
+            priv_key_d0 = bytearray([
+                0xcd, 0xb4, 0x57, 0xaf, 0x1c, 0x9f, 0x4c, 0x74,
+                0x02, 0x0c, 0x7e, 0x8b, 0xe9, 0x93, 0x3e, 0x28,
+                0x0c, 0xf0, 0x18, 0x0d, 0xf4, 0x6c, 0x0b, 0xda,
+                0x7a, 0xbb, 0xe6, 0x8f, 0xb7, 0xa0,0x45, 0x55
+            ])
+            priv_key_d1 = bytearray([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            ])
+            priv_key_d = priv_key_d0 + priv_key_d1
+            # Set two shares of the scalar secret_k
+            secret_k0 = bytearray([
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            ])
+            secret_k1 = bytearray([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+            ])
+            secret_k = secret_k0 + secret_k1
+        else:
+            raise RuntimeError('priv_key_d must be either 32B or 48B')
+                
+        # Create a clean array to keep the collected traces
         waves = np.array([])
-        # Loop to collect each section of the power trace
```

Comment:
Put this part into the `capture_ecdsa_sections()` function so that we can reuse this function for the new commands in capture.py

Reply:
Done.